spec/factories/users.rb

```diff
@@ -27,17 +33,47 @@
 #  verified                  :boolean
 #  created_at                :datetime         not null
 #  updated_at                :datetime         not null
+#  invited_by_id             :bigint
 #  zendesk_user_id           :bigint
 #
 # Indexes
 #
 #  index_users_on_email                 (email) UNIQUE
+#  index_users_on_invitation_token      (invitation_token) UNIQUE
+#  index_users_on_invitations_count     (invitations_count)
+#  index_users_on_invited_by_id         (invited_by_id)
 #  index_users_on_reset_password_token  (reset_password_token) UNIQUE
 #
+# Foreign Keys
+#
+#  fk_rails_...  (invited_by_id => users.id)
+#
 FactoryBot.define do
   factory :user do
     sequence(:uid)
     sequence(:email) { |n| "gary.gardengnome#{n}@example.green" }
     password { "userExamplePassword" }
+    name { "Gary Gnome" }
+
+    factory :admin_user do
+      role { "admin" }
+    end
+
+    factory :agent_user do
+      role { "agent" }
+    end
+
+    factory :invited_user do
+      association :invited_by, factory: :admin_user
+      invitation_created_at { 1.day.ago - 1.minute }
+      invitation_sent_at { 1.day.ago }
+      sequence(:invitation_token) do |n|
```

Comment:
Might not need the digest (though I know the tokens have to be different from each other)

Reply:
Might leave a comment explaining it's to overcome the uniqueness requirement

`# Creating a sequence because `invitation_token` is UNIQUE in the DB`